compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala

```diff
@@ -575,14 +575,24 @@ case class ProcessTranslator(cwlBundle: CwlBundle,
         param.name -> param.dxType
       }.toMap
 
+      val (instanceType, container, attributes, requirements) = availableDependencies
+        .collectFirst({
+          case (iname, i: Application) if iname == calleeName.getOrElse("") => {
+            (i.instanceType, i.container, i.attributes, i.requirements)
+          }
+        })
+        .getOrElse((DefaultInstanceType, NoImage, Vector.empty, Vector.empty))
+
       val applet = Application(
-          s"${wfName}_frag_${getStageId()}",
-          inputParams,
-          outputParams,
-          DefaultInstanceType,
-          NoImage,
-          ExecutableKindWfFragment(calleeName, blockPath, fqnDictTypes, scatterChunkSize),
-          standAloneWorkflow
+          name = s"${wfName}_frag_${getStageId()}",
+          inputs = inputParams,
+          outputs = outputParams,
+          instanceType = instanceType,
+          container = container,
+          kind = ExecutableKindWfFragment(calleeName, blockPath, fqnDictTypes, scatterChunkSize),
+          document = standAloneWorkflow,
+          attributes = attributes,
+          requirements = requirements
```

Comment:
In the previous integration tests two wdl tests could not pass, because the embedded task applet requires external network access but the frag does not have it.

I checked the implementation of applet/new https://github.com/dnanexus/nucleus/blob/db5a6120c97957c3c1a7c1c3d9625c4d32ea506c/apiserver/execution/applet._js#L32-L46
By default `network=[]`, so the frag will have None network access, while the inner one was given it during compilation
However, when running the applet, the network access is inherited from parent job:
https://github.com/dnanexus/nucleus/blob/dc128eed411a320ffae04952655008162c9307ac/apiserver/execution/exec_handlers/applets._js#L191-L204

So here I pulled the info from the callee applet and copied them to the frag applet.

Reply:
@commandlinegirl @Gvaihir

Reply:
Isn't this component responsible for CWL? How is this connected to WDL integration test?